src/services/dashboards/DashboardsLNB.vue

```diff
@@ -128,9 +89,81 @@ export default defineComponent({
             ]),
         });
 
-        const filterFavoriteItems = (menuItems: LNBItem[] = []) => {
+        const mashUpProjectGroup = (dashboardList: ProjectDashboardItem[] = []): LNBMenu[] => {
+            const noGroupList = [] as ProjectDashboardItem[];
+            const groupProjectDashboardMap = {} as Record<string, ProjectDashboardItem[]>;
+            // Since all lnbitem do not have groups, and even if they do have groups, they are not sorted.
+            // Items with groups are converted to map format, and if there are no groups, they are put in noGroupList.
+            dashboardList.forEach((d) => {
+                // With group
+                if (state.projectItems[d.project_id]) {
+                    if (groupProjectDashboardMap[state.projectItems[d.project_id].data.groupInfo.name]) {
+                        groupProjectDashboardMap[state.projectItems[d.project_id].data.groupInfo.name].push(d);
+                    } else {
+                        groupProjectDashboardMap[state.projectItems[d.project_id].data.groupInfo.name] = [d];
+                    }
+                }
+
+                // No-group
+                noGroupList.push(d);
```

Comment:
suggestion: It's pretty hard to follow 🥲
  
```suggestion
                // With group
                const groupId: string|undefined = state.projectItems[d.project_id]?.data.groupInfo.name;
                if (groupId) {
                   if (groupProjectDashboardMap[groupId]) {
                      groupProjectDashboardMap[groupId].push(d);
                  } else {
                      groupProjectDashboardMap[groupId] = [d];
                  }
                }
               
                // No-group
                noGroupList.push(d);
```

Reply:
Reflected!